### PR TITLE
ROX-18155: pg generic store: cleanup

### PR DESCRIPTION
**Pull request by janisz**
## Description

A detailed explanation of the changes in your PR.

Feel free to remove this section if it is overkill for your PR, and the title of your PR is sufficiently descriptive.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

TODO(replace-me)
Use this space to explain how you tested your PR, or, if you didn't test it, why
you did not do so. Valid reasons include, for example, "CI is sufficient",
"No testable changes". Feel free to attach JSON snippets, curl commands,
screenshots.

In addition to reviewing your code, reviewers **must** also review your testing
instructions and make sure they are sufficient.

### CODE REVIEW

**Comment by janisz** [comment]
Current dependencies on/for this PR:
* master
  * **PR #6766** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/6766" target="_blank">Graphite</a> 
    * **PR #6767** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/6767" target="_blank">Graphite</a> 
      * **PR #6768** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/6768" target="_blank">Graphite</a> 
        * **PR #6769** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/6769" target="_blank">Graphite</a> 
          * **PR #6770** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/6770" target="_blank">Graphite</a> 
            * **PR #6771** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/6771" target="_blank">Graphite</a> 
              * **PR #6772** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/6772" target="_blank">Graphite</a> 
                * **PR #6773** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/6773" target="_blank">Graphite</a> 
                  * **PR #6774** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/6774" target="_blank">Graphite</a> 
                    * **PR #6775** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/6775" target="_blank">Graphite</a> 
                      * **PR #6776** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/6776" target="_blank">Graphite</a> 
                        * **PR #6777** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/6777" target="_blank">Graphite</a> 
                          * **PR #6778** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/6778" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/stackrox/stackrox/6778?utm_source=stack-comment).

File: tools/generate-helpers/pg-table-bindings/store.go.tpl [hunk 1]
```diff
@@ -79,11 +77,13 @@ var (
     {{- end }}
 )
 
-// Store is the interface to interact with the storage for {{.Type}}
+type storeType = {{ .Type }}
+
+// Store is the interface to interact with the storage for storeType
```
Comment (rhybrillou) [inline tools/generate-helpers/pg-table-bindings/store.go.tpl:82]: I think you should stick to {{.Type}} or {{.TrimmedType}} in that comment.

File: tools/generate-helpers/pg-table-bindings/store.go.tpl [hunk 2]
```diff
@@ -136,16 +134,16 @@ func New(db postgres.DB) Store {
...
 
-func pkGetter(obj *{{ .Type }}) {{$singlePK.Type}} {
+func pkGetter(obj *storeType) {{$singlePK.Type}} {
     return {{ $singlePK.Getter "obj" }}
 }
 
-func metricsSetPostgresOperationDurationTime(start time.Time, op ops.Op) {
-    metrics.SetPostgresOperationDurationTime(start, op, storeName)
+func metricsSetPostgresOperationDurationTime(op ops.Op) {
+    metrics.SetPostgresOperationDurationTime(time.Now(), op, storeName)
```
[the remark this answers is not in the document]
Comment (janisz) [reply tools/generate-helpers/pg-table-bindings/store.go.tpl:142]: Ouch, good point. I'll revert that

**Review by rhybrillou — APPROVED** [review]

**Comment by janisz** [comment]
@janisz started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/stackrox/stackrox/6778).

**Comment by janisz** [comment]
@janisz merged this pull request with [Graphite](https://app.graphite.dev/github/pr/stackrox/stackrox/6778).

**Pull request merged**